### PR TITLE
:memo: Add leap approaches

### PR DESCRIPTION
**Pull request by jonmcalder**
I accidentally nuked my WIP on this (uncommitted changes) while switching between  branches and rebasing so I had to redo it. Please could you review and check for any issues.

I've opted not to include a performance article for now, but have hinted at benchmarking results for different approaches and the tradeoffs between non-vectorised and vectorised implementations.

### CODE REVIEW

**Comment by jonmcalder** [comment]
Other questions / consisderations:

I opted for a "conditional-expression" approach like [Haskell has](https://exercism.org/tracks/haskell/exercises/leap/approaches/conditional-expression) rather than "ternary-operator" since R doesn't really have a ternary operator for `if () ... else ...` though `ifelse()` is essentially a vectorised ternary operator.

Do you think it is worth including "cheat" approaches for [lubridate::leap_year](https://lubridate.tidyverse.org/reference/leap_year.html) and/or a {lubridate} equivalent for [datetime addition](https://exercism.org/tracks/python/exercises/leap/approaches/datetime-addition)?

**Comment by colinleach** [comment]
> I accidentally nuked my WIP on this (uncommitted changes) while switching between branches and rebasing

You have my sympathy, we've all been there.

> Do you think it is worth including "cheat" approaches for [lubridate::leap_year](https://lubridate.tidyverse.org/reference/leap_year.html) and/or a {lubridate} equivalent for [datetime addition](https://exercism.org/tracks/python/exercises/leap/approaches/datetime-addition)?

Yes, I think it's good to cover all the options in approaches, including things that go beyond what's allowed in Exercism. It's all part of learning about the language, even if student has to go elsewhere to practice. Need to include a warning in the notes, though.

File: exercises/practice/leap/.approaches/introduction.md
Comment (colinleach) [inline exercises/practice/leap/.approaches/introduction.md:1]: `[conditional expression][conditional-expression]` at line 5 is a broken link, needs a reference at the bottom of the file.

**Comment by colinleach** [comment]
One broken link in `.approaches/introduction.md`, otherwise it looks great.

**Review by colinleach — APPROVED** [review]

**Comment by jonmcalder** [comment]
> You have my sympathy, we've all been there.

Haha - thanks! 🤗 

> Yes, I think it's good to cover all the options in approaches, including things that go beyond what's allowed in Exercism. It's all part of learning about the language, even if student has to go elsewhere to practice. Need to include a warning in the notes, though.

Ok sure, will add one or two {lubridate} based approaches in a later PR.

**Pull request merged**